sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificatesUtils.java

```diff
@@ -38,6 +38,26 @@ private static String getIdMetadataPiece(String[] pieces, int index) {
         return pieces.length >= index + 1 ? pieces[index] : null;
     }
 
+    public static String toHexString(byte[] x509Thumbprint) {
+        if (x509Thumbprint == null) {
+            return null;
+        }
+
+        StringBuilder hexString = new StringBuilder();
+
+        for (byte b : x509Thumbprint) {
+            String hex = Integer.toHexString(0xFF & b);
+
+            if (hex.length() == 1) {
+                hexString.append('0');
```

Comment:
Shouldn't you be prepending? I.e. likely an insert into index 0? But then that leads into my next comment below...